### ISSUE TITLE
Knowledge Engine Docker Image in registry for local development

### ISSUE THREAD

**Issue by Sophietje** [TNO/knowledge-engine#227]
In GitLab by @jarne.kerkaert.inetum-realdolmen.world on May 27, 2021, 12:24

It would be very beneficial if there were a **docker image** with the latest version of the Knowledge Engine available in a registry somewhere. This could then be used for very easily managing a local Knowledge Engine instance.

I also want to add that simply having a Dockerfile and docker compose file available would already be a big help!

**Comment by Sophietje** [comment]
In GitLab by @jarne.kerkaert.inetum-realdolmen.world on May 31, 2021, 09:48

@barry.nouwt.tno.nl 

I'm wondering if it would be all right for us to create a Dockerfile ourselves and maintain that in our own docker registry for now? This is a **private Azure Registry**, so it would only be accessible to us. Is there some kind of license that forbids that?

Same question for the Generic Adapter docker image!

Thanks in advance!

**Comment by Sophietje** [comment]
In GitLab by @aleksandar.tomcic.vizlore.com on May 31, 2021, 10:49

Hi Jarne, 
For Generic Adapter there is not anything that forbids it.
Maybe, at some point, you can share a Dockerfile with us? 

Regards, 
Aleksandar.

**Comment by Sophietje** [comment]
In GitLab by @barry.nouwt.tno.nl on May 31, 2021, 10:52

No problem to create your own Jarne. And as Aleksandar suggests, if you want to contribute it to InterConnect that would be awesome!

**Comment by Sophietje** [comment]
In GitLab by @jarne.kerkaert.inetum-realdolmen.world on May 31, 2021, 13:58

@barry.nouwt.tno.nl @aleksandar.tomcic.vizlore.com 

Okay that's great! I will make a PR once I get something working. 

As for the Generic Adapter, I thought there was already a Dockerfile? Or is that meant for different purposes?

**Comment by Sophietje** [comment]
In GitLab by @aleksandar.tomcic.vizlore.com on May 31, 2021, 15:15

Yes, there is already Dockerfile for Generic Adapter, but we are still working on it i.e. it is not finished.
You can use it, there is also documentation for that, and it you have any comments or suggestions, contact us.

Regards,
Aleksandar

**Comment by Sophietje** [comment]
In GitLab by @jarne.kerkaert.inetum-realdolmen.world on Jun 1, 2021, 15:29

Hi @barry.nouwt.tno.nl,

I've created a PR [here](https://gitlab.inesctec.pt/interconnect/knowledge-engine/-/merge_requests/164). I had to add the maven dependency plugin to the pom in order to get it working locally. I also added some things to the .gitignore, as I use a different IDE. I hope that's okay!

Some things could still be improved, such as extracting the image version to a variable somewhere. But I think it's a good start!
